static/app/utils/useOverlay.tsx

```diff
@@ -11,6 +12,63 @@ import {mergeProps} from '@react-aria/utils';
 import {useOverlayTriggerState} from '@react-stately/overlays';
 import {OverlayTriggerProps as OverlayTriggerStateProps} from '@react-types/overlays';
 
+type PreventOverflowOptions = NonNullable<typeof preventOverflow['options']>;
+
+/**
+ * PopperJS modifier to change the popper element's width/height to prevent
+ * overflowing. Based on
+ * https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+ */
+const maxSize: Modifier<'maxSize', PreventOverflowOptions> = {
```

Comment:
For now while we're still on `popperjs` we'll have to implement this manually. Once we switch over to Floating UI, we can use it's native `size` middleware.